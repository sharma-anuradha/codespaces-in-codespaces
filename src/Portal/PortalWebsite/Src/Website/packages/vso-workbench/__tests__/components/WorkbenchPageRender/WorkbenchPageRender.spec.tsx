import React from 'react';
import renderer from 'react-test-renderer';

import { EnvironmentStateInfo } from 'vso-client-core';

import { WorkbenchPageRender } from '../../../src/react-app/components/WorkbenchPage/WorkbenchPageRender';
import { EnvironmentWorkspaceState } from '../../../src/interfaces/EnvironmentWorkspaceState';

describe('WorkbenchPageRender', () => {
    it('should render WorkbenchPageRender environment states', () => {
        let i = 0;
        for (let state in EnvironmentStateInfo) {
            const message = `${i++}`;

            const component = renderer.create(
                <WorkbenchPageRender
                    environmentInfo={null}
                    environmentState={state as EnvironmentStateInfo}
                    message={message}
                    startEnvironment={() => {}}
                    handleAPIError={() => {}}
                    onSignIn={() => {}}
                />
            );

            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        }
    });

    it('should render WorkbenchPageRender workbench states', () => {
        let i = 0;
        for (let state in EnvironmentWorkspaceState) {
            const message = `${i++}`;

            const component = renderer.create(
                <WorkbenchPageRender
                    environmentInfo={null}
                    environmentState={state as EnvironmentWorkspaceState}
                    message={message}
                    startEnvironment={() => {}}
                    handleAPIError={() => {}}
                    onSignIn={() => {}}
                />
            );

            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
