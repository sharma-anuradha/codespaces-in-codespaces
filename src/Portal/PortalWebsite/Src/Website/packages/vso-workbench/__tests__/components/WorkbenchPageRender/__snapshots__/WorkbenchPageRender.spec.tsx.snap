// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkbenchPageRender should render WorkbenchPageRender environment states 1`] = `
<div
  className="vsonline-splash-screen-main "
>
  <div
    className="vsonline-splash-screen-extensions-pane"
  />
  <div
    className="vsonline-splash-screen-tree-pane"
  />
  <div
    className="vsonline-splash-screen-editor"
  >
    <div
      className="vsonline-splash-screen-titlebar"
    >
      <div
        className="vsonline-splash-screen-titlebar-tab"
      />
    </div>
    <div
      className="vsonline-splash-screen-body"
    >
      <div
        className="vso-splash-screen__block"
      >
        <div
          className="vso-splash-screen__message"
        >
           
          The workspace has been deleted.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkbenchPageRender should render WorkbenchPageRender environment states 2`] = `
Array [
  <div
    className="vsonline-splash-screen-main "
  >
    <div
      className="vsonline-splash-screen-extensions-pane"
    />
    <div
      className="vsonline-splash-screen-tree-pane"
    />
    <div
      className="vsonline-splash-screen-editor"
    >
      <div
        className="vsonline-splash-screen-titlebar"
      >
        <div
          className="vsonline-splash-screen-titlebar-tab"
        />
      </div>
      <div
        className="vsonline-splash-screen-body"
      >
        <div
          className="vso-splash-screen__block"
        >
          <div
            className="vso-splash-screen__message"
          >
            <div
              className="vso-ms-spinner  "
            >
              <div
                className="vso-ms-spinner__rect vso-ms-spinner__rect--1"
              />
              <div
                className="vso-ms-spinner__rect vso-ms-spinner__rect--2"
              />
              <div
                className="vso-ms-spinner__rect vso-ms-spinner__rect--3"
              />
              <div
                className="vso-ms-spinner__rect vso-ms-spinner__rect--4"
              />
            </div>
             
            Connecting...
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="vso-workbench-root"
    id="js-vscode-workbench-placeholder"
  />,
]
`;

exports[`WorkbenchPageRender should render WorkbenchPageRender environment states 3`] = `
<div
  className="vsonline-splash-screen-main "
>
  <div
    className="vsonline-splash-screen-extensions-pane"
  />
  <div
    className="vsonline-splash-screen-tree-pane"
  />
  <div
    className="vsonline-splash-screen-editor"
  >
    <div
      className="vsonline-splash-screen-titlebar"
    >
      <div
        className="vsonline-splash-screen-titlebar-tab"
      />
    </div>
    <div
      className="vsonline-splash-screen-body"
    >
      <div
        className="vso-splash-screen__block"
      >
        <div
          className="vso-splash-screen__message"
        >
           
          The workspace is not available.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkbenchPageRender should render WorkbenchPageRender environment states 4`] = `
<div
  className="vsonline-splash-screen-main "
>
  <div
    className="vsonline-splash-screen-extensions-pane"
  />
  <div
    className="vsonline-splash-screen-tree-pane"
  />
  <div
    className="vsonline-splash-screen-editor"
  >
    <div
      className="vsonline-splash-screen-titlebar"
    >
      <div
        className="vsonline-splash-screen-titlebar-tab"
      />
    </div>
    <div
      className="vsonline-splash-screen-body"
    >
      <div
        className="vso-splash-screen__block"
      >
        <div
          className="vso-splash-screen__message"
        >
           
          The workspace is shutdown.
        </div>
        <div
          className="vso-splash-screen__button"
        >
          <span
            className="vso-button "
            onClick={[Function]}
          >
            Connect
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkbenchPageRender should render WorkbenchPageRender environment states 5`] = `
<div
  className="vsonline-splash-screen-main "
>
  <div
    className="vsonline-splash-screen-extensions-pane"
  />
  <div
    className="vsonline-splash-screen-tree-pane"
  />
  <div
    className="vsonline-splash-screen-editor"
  >
    <div
      className="vsonline-splash-screen-titlebar"
    >
      <div
        className="vsonline-splash-screen-titlebar-tab"
      />
    </div>
    <div
      className="vsonline-splash-screen-body"
    >
      <div
        className="vso-splash-screen__block"
      >
        <div
          className="vso-splash-screen__message"
        >
          <div
            className="vso-ms-spinner  "
          >
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--1"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--2"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--3"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--4"
            />
          </div>
           
          The workspace is shutting down.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkbenchPageRender should render WorkbenchPageRender environment states 6`] = `
<div
  className="vsonline-splash-screen-main "
>
  <div
    className="vsonline-splash-screen-extensions-pane"
  />
  <div
    className="vsonline-splash-screen-tree-pane"
  />
  <div
    className="vsonline-splash-screen-editor"
  >
    <div
      className="vsonline-splash-screen-titlebar"
    >
      <div
        className="vsonline-splash-screen-titlebar-tab"
      />
    </div>
    <div
      className="vsonline-splash-screen-body"
    >
      <div
        className="vso-splash-screen__block"
      >
        <div
          className="vso-splash-screen__message"
        >
           
          The workspace failed.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkbenchPageRender should render WorkbenchPageRender environment states 7`] = `
<div
  className="vsonline-splash-screen-main "
>
  <div
    className="vsonline-splash-screen-extensions-pane"
  />
  <div
    className="vsonline-splash-screen-tree-pane"
  />
  <div
    className="vsonline-splash-screen-editor"
  >
    <div
      className="vsonline-splash-screen-titlebar"
    >
      <div
        className="vsonline-splash-screen-titlebar-tab"
      />
    </div>
    <div
      className="vsonline-splash-screen-body"
    >
      <div
        className="vso-splash-screen__block"
      >
        <div
          className="vso-splash-screen__message"
        >
          <div
            className="vso-ms-spinner  "
          >
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--1"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--2"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--3"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--4"
            />
          </div>
           
          Starting the workspace...
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkbenchPageRender should render WorkbenchPageRender environment states 8`] = `
<div
  className="vsonline-splash-screen-main "
>
  <div
    className="vsonline-splash-screen-extensions-pane"
  />
  <div
    className="vsonline-splash-screen-tree-pane"
  />
  <div
    className="vsonline-splash-screen-editor"
  >
    <div
      className="vsonline-splash-screen-titlebar"
    >
      <div
        className="vsonline-splash-screen-titlebar-tab"
      />
    </div>
    <div
      className="vsonline-splash-screen-body"
    >
      <div
        className="vso-splash-screen steps"
      >
        <h4>
          Preparing your 
          Visual Studio Codespace
        </h4>
        <ul>
          <li>
            <div
              className="step-header"
            >
              <span
                className="status status-icon-progress status-icon-black"
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fillRule="nonzero"
                    id="progress"
                    transform="translate(1.000000, 0.000000)"
                  >
                    <path
                      d="M7.25925926,0.870789 C7.43750519,0.870789 7.60493481,0.905896852 7.76157926,0.976112556 C7.91821333,1.04092633 8.05594222,1.13274559 8.17476593,1.25157967 C8.2936,1.37040337 8.38541926,1.50813226 8.45023407,1.66476633 C8.52045185,1.82141078 8.55555556,1.98884041 8.55555556,2.16708633 C8.55555556,2.34532189 8.52045185,2.51276189 8.45023407,2.66939596 C8.38541926,2.82604041 8.2936,2.9637693 8.17476593,3.082593 C8.05594222,3.2014167 7.91821333,3.29594263 7.76157926,3.36616041 C7.60493481,3.43097522 7.43750519,3.46338263 7.25925926,3.46338263 C7.08101333,3.46338263 6.9135837,3.43097522 6.75693926,3.36616041 C6.60030519,3.29594263 6.4625763,3.2014167 6.34375259,3.082593 C6.22491852,2.9637693 6.13040296,2.82604041 6.06018519,2.66939596 C5.99537037,2.51276189 5.96296296,2.34532189 5.96296296,2.16708633 C5.96296296,1.98884041 5.99537037,1.82141078 6.06018519,1.66476633 C6.13040296,1.50813226 6.22491852,1.37040337 6.34375259,1.25157967 C6.4625763,1.13274559 6.60030519,1.04092633 6.75693926,0.976112556 C6.9135837,0.905896852 7.08101333,0.870789 7.25925926,0.870789 Z M1.69328444,3.99000041 C1.69328444,3.82796337 1.72299556,3.67673226 1.78240741,3.5362967 C1.84181926,3.39586115 1.92284296,3.27434115 2.02545778,3.17171596 C2.13348593,3.06368781 2.25771259,2.97996781 2.39814815,2.92055596 C2.5439763,2.85574115 2.69791407,2.82333374 2.85995111,2.82333374 C3.02198815,2.82333374 3.17322963,2.85574115 3.31365481,2.92055596 C3.45409037,2.97996781 3.57562074,3.06368781 3.67824593,3.17171596 C3.7862637,3.27434115 3.8699837,3.39586115 3.92939556,3.5362967 C3.99421037,3.67673226 4.02661778,3.82796337 4.02661778,3.99000041 C4.02661778,4.15203744 3.99421037,4.30597522 3.92939556,4.45180337 C3.8699837,4.59223893 3.7862637,4.71646559 3.67824593,4.82449374 C3.57562074,4.92711893 3.45409037,5.00813226 3.31365481,5.06754411 C3.17322963,5.12696633 3.02198815,5.15666707 2.85995111,5.15666707 C2.69791407,5.15666707 2.5439763,5.12696633 2.39814815,5.06754411 C2.25771259,5.00813226 2.13348593,4.92711893 2.02545778,4.82449374 C1.92284296,4.71646559 1.84181926,4.59223893 1.78240741,4.45180337 C1.72299556,4.30597522 1.69328444,4.15203744 1.69328444,3.99000041 Z M1.03703704,7.35227152 C1.18286519,7.35227152 1.31789778,7.37927596 1.44212444,7.43328485 C1.56636148,7.48730411 1.67437926,7.56291448 1.76619852,7.6601367 C1.86343111,7.75196633 1.93904148,7.85998411 1.99306074,7.98421078 C2.04706963,8.10844781 2.07407407,8.24347004 2.07407407,8.38930856 C2.07407407,8.5351367 2.04706963,8.6701693 1.99306074,8.79439596 C1.93904148,8.918633 1.86343111,9.02935744 1.76619852,9.12657967 C1.67437926,9.21839893 1.56636148,9.291313 1.44212444,9.34532189 C1.31789778,9.39934115 1.18286519,9.42634559 1.03703704,9.42634559 C0.891203704,9.42634559 0.756173185,9.39934115 0.631944444,9.34532189 C0.507715704,9.291313 0.396990222,9.21839893 0.299768,9.12657967 C0.207947704,9.02935744 0.135030519,8.918633 0.0810185185,8.79439596 C0.0270062074,8.6701693 0,8.5351367 0,8.38930856 C0,8.24347004 0.0270062074,8.10844781 0.0810185185,7.98421078 C0.135030519,7.85998411 0.207947704,7.75196633 0.299768,7.6601367 C0.396990222,7.56291448 0.507715704,7.48730411 0.631944444,7.43328485 C0.756173185,7.37927596 0.891203704,7.35227152 1.03703704,7.35227152 Z M1.9525437,12.7886374 C1.9525437,12.6644004 1.97685185,12.5482523 2.02545778,12.440193 C2.07407407,12.3268449 2.13888889,12.2295708 2.21990222,12.1485782 C2.30092593,12.0675856 2.39545185,12.0027708 2.50346963,11.9541337 C2.61690074,11.9054967 2.73572444,11.88123 2.85995111,11.88123 C2.98417778,11.88123 3.10030519,11.9054967 3.20833333,11.9541337 C3.32175407,12.0027708 3.41898667,12.0675856 3.5,12.1485782 C3.58101333,12.2295708 3.64582815,12.3268449 3.69444444,12.440193 C3.74306074,12.5482523 3.76735852,12.6644004 3.76735852,12.7886374 C3.76735852,12.9128745 3.74306074,13.0317189 3.69444444,13.1450671 C3.64582815,13.2531263 3.58101333,13.3476004 3.5,13.4286967 C3.41898667,13.5096893 3.32175407,13.5745041 3.20833333,13.6231411 C3.10030519,13.6716745 2.98417778,13.6960449 2.85995111,13.6960449 C2.73572444,13.6960449 2.61690074,13.6716745 2.50346963,13.6231411 C2.39545185,13.5745041 2.30092593,13.5096893 2.21990222,13.4286967 C2.13888889,13.3476004 2.07407407,13.2531263 2.02545778,13.1450671 C1.97685185,13.0317189 1.9525437,12.9128745 1.9525437,12.7886374 Z M7.25925926,13.8337634 C7.47530519,13.8337634 7.65895407,13.9093634 7.81018519,14.0605634 C7.9614163,14.2118671 8.03703704,14.3955263 8.03703704,14.6115411 C8.03703704,14.827556 7.9614163,15.0112152 7.81018519,15.1624152 C7.65895407,15.3137189 7.47530519,15.3893189 7.25925926,15.3893189 C7.04321333,15.3893189 6.85956444,15.3137189 6.70833333,15.1624152 C6.55710222,15.0112152 6.48148148,14.827556 6.48148148,14.6115411 C6.48148148,14.3955263 6.55710222,14.2118671 6.70833333,14.0605634 C6.85956444,13.9093634 7.04321333,13.8337634 7.25925926,13.8337634 Z M11.0104296,12.7886374 C11.0104296,12.6049782 11.0725481,12.4510819 11.1967852,12.3268449 C11.3210222,12.2026078 11.4749185,12.1404893 11.6585778,12.1404893 C11.842237,12.1404893 11.9961333,12.2026078 12.1203704,12.3268449 C12.2446074,12.4510819 12.3067259,12.6049782 12.3067259,12.7886374 C12.3067259,12.9722967 12.2446074,13.126193 12.1203704,13.25043 C11.9961333,13.3746671 11.842237,13.4367856 11.6585778,13.4367856 C11.4749185,13.4367856 11.3210222,13.3746671 11.1967852,13.25043 C11.0725481,13.126193 11.0104296,12.9722967 11.0104296,12.7886374 Z M13.4814815,7.87079004 C13.6218963,7.87079004 13.743437,7.92210263 13.8461037,8.02472781 C13.9486667,8.12734263 14,8.248873 14,8.38930856 C14,8.52974411 13.9486667,8.65126411 13.8461037,8.7538893 C13.743437,8.85651448 13.6218963,8.90782707 13.4814815,8.90782707 C13.3410667,8.90782707 13.2195259,8.85651448 13.1168593,8.7538893 C13.0142963,8.65126411 12.962963,8.52974411 12.962963,8.38930856 C12.962963,8.248873 13.0142963,8.12734263 13.1168593,8.02472781 C13.2195259,7.92210263 13.3410667,7.87079004 13.4814815,7.87079004 Z M11.6585778,2.56407448 C11.8530222,2.56407448 12.0366815,2.60188485 12.2094519,2.67750559 C12.3823259,2.75311596 12.5335259,2.85574115 12.6631556,2.98537078 C12.7927852,3.11500041 12.8954519,3.26623152 12.9710519,3.43907448 C13.0466519,3.61191744 13.0845037,3.79555596 13.0845037,3.99000041 C13.0845037,4.18444485 13.0466519,4.36808337 12.9710519,4.54092633 C12.8954519,4.7137693 12.7927852,4.86500041 12.6631556,4.99463004 C12.5335259,5.12425967 12.3823259,5.22688485 12.2094519,5.30250559 C12.0366815,5.37811596 11.8530222,5.41592633 11.6585778,5.41592633 C11.4641333,5.41592633 11.2804741,5.37811596 11.1076,5.30250559 C10.9348296,5.22688485 10.7835259,5.12425967 10.6538963,4.99463004 C10.5242667,4.86500041 10.4217037,4.7137693 10.3460622,4.54092633 C10.2704519,4.36808337 10.2326415,4.18444485 10.2326415,3.99000041 C10.2326415,3.79555596 10.2704519,3.61191744 10.3460622,3.43907448 C10.4217037,3.26623152 10.5242667,3.11500041 10.6538963,2.98537078 C10.7835259,2.85574115 10.9348296,2.75311596 11.1076,2.67750559 C11.2804741,2.60188485 11.4641333,2.56407448 11.6585778,2.56407448 Z"
                      id="Shape"
                    />
                  </g>
                </svg>
              </span>
              <span
                className="title"
              >
                Initialize   Codespace
              </span>
            </div>
          </li>
        </ul>
        <button
          className="connect-button"
          onClick={[Function]}
        >
          Connect
        </button>
        <style>
          
.vso-splash-screen ul {
  list-style: none;
}

.vso-splash-screen * {
  box-sizing: border-box;
}

.vso-splash-screen h4 { font-weight: 600; }

@keyframes spin {
  0% {
      transform: rotate(0deg);
  }
  100% {
      transform: rotate(360deg);
  }
}

.steps {
  grid-column: 4;
  grid-row: 2;
  padding: 20px 14px;
  font-family: var(--vscode-font-family, -apple-system, system-ui, 'Segoe UI', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji');
  font-size: var(--vscode-font-size, 13px);
  overflow: auto;
}

.is-vs-codespaces .steps {
  background: var(--vscode-editor-background, rgb(30, 30, 30));
}

.is-vs-codespaces h4 {
  color: var(--vscode-editor-foreground, #cccccc);
}

.steps li {
  background: var(--vscode-panel-background, #FAFBFD);
  border: 1px solid var(--vscode-panel-border, #DDE0E4);
  padding: 14px 0 0;
  margin-top: 14px;
}

.is-vs-codespaces .steps li {
  background: var(--vscode-panel-background, #181818);
  border: 1px solid var(--vscode-panel-border, #444444);
}

.step-header {
  display: flex;
  margin-bottom: 14px;
  padding-left: 14px;
}

.step-header .title {
  font-weight: 500;
  flex-grow: 1;
  padding-left: 10px;
}

.is-vs-codespaces .step-header .title {
  color: var(--vscode-editor-foreground, #cccccc);
}

.step-log {
  font-family: SFMono-Regular, Consolas, 'Liberation Mono', Menlo, monospace;
  font-size: 12px;
  line-height: 1.4;
  overflow: hidden;
  flex-direction: column-reverse;
  border-top: 1px solid var(--vscode-panel-border, #DDE0E4);
  padding-left: 15px;
  white-space: nowrap;
  display: none;
}

.is-vs-codespaces .step-log {
  border-top: 1px solid var(--vscode-panel-border, #444444);
}

.is-active {
  display: flex;
}

.status {
  width: 16px;
  height: 16px;
  position: relative;
  display: inline-block;
  vertical-align: middle;
}

.status-icon-blue { fill: var(--vscode-debugIcon-continueForeground, #0366D6); }
.status-icon-green { fill: var(--vscode-debugIcon-startForeground, #28A745); }
.status-icon-red { fill: var(--vscode-errorForeground, #CB2431); }
.status-icon-black { fill: var(--vscode-editor-foreground, #444D56); }

.status-icon-progress {
  animation-name: spin;
  animation-duration: 1s;
  animation-iteration-count: infinite;
  animation-timing-function: linear;
}

.expanded {
  transform: rotate(180deg);
}

.step-header .action {
  font-weight: 600;
  padding-right: 15px;
}

.action button {
  border: 0;
  background: none;
  cursor: pointer;
}

.vsonline-notification {
  font-size: 14px;
  background: var(--vscode-notifications-background, #FAFBFD);
  padding: 14px 36px 14px 14px;
  box-shadow: 0px 0px 6px rgba(0,0,0,0.4);
  color: var(--vscode-notifications-foreground, black);
  position: absolute;
  bottom: 42px;
  right: 20px;
  max-width: 300px;
  opacity: 0;
  transform: translateY(20px);
  transition: all 0.3s ease-in-out;
}

.vsonline-notification button {
  position: absolute;
  border: 0;
  background: none;
  cursor: pointer;
  right: 14px;
  top: 14px;
}

.vsonline-notification.shown {
  transform: translateY(0);
  opacity: 1;
}

.is-vs-codespaces .vsonline-notification {
  background: var(--vscode-notifications-background, #181818);
  color: var(--vscode-notifications-foreground, #CCCCCC);
}

.terminal-container {
  position: relative;
  overflow: hidden;
  height: 102px;
  margin: 10px;
}

.connect-button {
  display: inline;
  float: left;
  background: var(--vscode-button-background, rgb(0, 122, 204));
  color: var(--vscode-button-foreground, white);
  border: none;
  margin-top: 20px;
  padding: 5px 10px;
  font-size: 12px;
  display: none;
}

.connect-button.ready {
  display: block;
}

.codespaces-terminal {
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
}

/**
 * Copyright (c) 2014 The xterm.js authors. All rights reserved.
 * Copyright (c) 2012-2013, Christopher Jeffrey (MIT License)
 * https://github.com/chjj/term.js
 * @license MIT
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 *
 * Originally forked from (with the author's permission):
 *   Fabrice Bellard's javascript vt100 for jslinux:
 *   http://bellard.org/jslinux/
 *   Copyright (c) 2011 Fabrice Bellard
 *   The original design remains. The terminal itself
 *   has been extended to include xterm CSI codes, among
 *   other features.
 */

/**
 *  Default styles for xterm.js
 */

 .xterm {
  font-feature-settings: "liga" 0;
  user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.xterm.focus,
.xterm:focus {
  outline: none;
}

.xterm .xterm-helpers {
  position: absolute;
  top: 0;
  /**
   * The z-index of the helpers must be higher than the canvases in order for
   * IMEs to appear on top.
   */
  z-index: 5;
}

.xterm .xterm-helper-textarea {
  /*
   * HACK: to fix IE's blinking cursor
   * Move textarea out of the screen to the far left, so that the cursor is not visible.
   */
  position: absolute;
  opacity: 0;
  left: -9999em;
  top: 0;
  width: 0;
  height: 0;
  z-index: -5;
  /** Prevent wrapping so the IME appears against the textarea at the correct position */
  white-space: nowrap;
  overflow: hidden;
  resize: none;
}

.xterm .composition-view {
  /* TODO: Composition position got messed up somewhere */
  background: #000;
  color: #FFF;
  display: none;
  position: absolute;
  white-space: nowrap;
  z-index: 1;
}

.xterm .composition-view.active {
  display: block;
}

.xterm .xterm-viewport {
  /* On OS X this is required in order for the scroll bar to appear fully opaque */
  background-color: #000;
  overflow-y: scroll;
  cursor: default;
  position: absolute;
  right: 0;
  left: 0;
  top: 0;
  bottom: 0;
}

.xterm .xterm-screen {
  position: relative;
}

.xterm .xterm-screen canvas {
  position: absolute;
  left: 0;
  top: 0;
}

.xterm .xterm-scroll-area {
  visibility: hidden;
}

.xterm-char-measure-element {
  display: inline-block;
  visibility: hidden;
  position: absolute;
  top: 0;
  left: -9999em;
  line-height: normal;
}

.xterm {
  cursor: text;
}

.xterm.enable-mouse-events {
  /* When mouse events are enabled (eg. tmux), revert to the standard pointer cursor */
  cursor: default;
}

.xterm.xterm-cursor-pointer {
  cursor: pointer;
}

.xterm.column-select.focus {
  /* Column selection mode */
  cursor: crosshair;
}

.xterm .xterm-accessibility,
.xterm .xterm-message {
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
  z-index: 10;
  color: transparent;
}

.xterm .live-region {
  position: absolute;
  left: -9999px;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.xterm-dim {
  opacity: 0.5;
}

.xterm-underline {
  text-decoration: underline;
}
        </style>
        <div
          className="vsonline-notification"
        >
          <p
            className="text"
          >
            
          </p>
          <button
            onClick={[Function]}
          >
            <span
              className="status status-icon-black"
            >
              <svg
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.72 3.72C3.86062 3.57955 4.05125 3.50066 4.25 3.50066C4.44875 3.50066 4.63937 3.57955 4.78 3.72L8 6.94L11.22 3.72C11.2887 3.64631 11.3715 3.58721 11.4635 3.54622C11.5555 3.50523 11.6548 3.48319 11.7555 3.48141C11.8562 3.47963 11.9562 3.49816 12.0496 3.53588C12.143 3.5736 12.2278 3.62974 12.299 3.70096C12.3703 3.77218 12.4264 3.85702 12.4641 3.9504C12.5018 4.04379 12.5204 4.14382 12.5186 4.24452C12.5168 4.34523 12.4948 4.44454 12.4538 4.53654C12.4128 4.62854 12.3537 4.71134 12.28 4.78L9.06 8L12.28 11.22C12.3537 11.2887 12.4128 11.3715 12.4538 11.4635C12.4948 11.5555 12.5168 11.6548 12.5186 11.7555C12.5204 11.8562 12.5018 11.9562 12.4641 12.0496C12.4264 12.143 12.3703 12.2278 12.299 12.299C12.2278 12.3703 12.143 12.4264 12.0496 12.4641C11.9562 12.5018 11.8562 12.5204 11.7555 12.5186C11.6548 12.5168 11.5555 12.4948 11.4635 12.4538C11.3715 12.4128 11.2887 12.3537 11.22 12.28L8 9.06L4.78 12.28C4.63782 12.4125 4.44977 12.4846 4.25547 12.4812C4.06117 12.4777 3.87579 12.399 3.73837 12.2616C3.60096 12.1242 3.52225 11.9388 3.51882 11.7445C3.51539 11.5502 3.58752 11.3622 3.72 11.22L6.94 8L3.72 4.78C3.57955 4.63938 3.50066 4.44875 3.50066 4.25C3.50066 4.05125 3.57955 3.86063 3.72 3.72Z"
                  fillRule="evenodd"
                  logp-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkbenchPageRender should render WorkbenchPageRender workbench states 1`] = `
<div
  className="vsonline-splash-screen-main "
>
  <div
    className="vsonline-splash-screen-extensions-pane"
  />
  <div
    className="vsonline-splash-screen-tree-pane"
  />
  <div
    className="vsonline-splash-screen-editor"
  >
    <div
      className="vsonline-splash-screen-titlebar"
    >
      <div
        className="vsonline-splash-screen-titlebar-tab"
      />
    </div>
    <div
      className="vsonline-splash-screen-body"
    >
      <div
        className="vso-splash-screen__block"
      >
        <div
          className="vso-splash-screen__message"
        >
           
          Please sign in to proceed.
        </div>
        <div
          className="vso-splash-screen__button"
        >
          <span
            className="vso-button "
            onClick={[Function]}
          >
            Sign in
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkbenchPageRender should render WorkbenchPageRender workbench states 2`] = `
<div
  className="vsonline-splash-screen-main "
>
  <div
    className="vsonline-splash-screen-extensions-pane"
  />
  <div
    className="vsonline-splash-screen-tree-pane"
  />
  <div
    className="vsonline-splash-screen-editor"
  >
    <div
      className="vsonline-splash-screen-titlebar"
    >
      <div
        className="vsonline-splash-screen-titlebar-tab"
      />
    </div>
    <div
      className="vsonline-splash-screen-body"
    >
      <div
        className="vso-splash-screen__block"
      >
        <div
          className="vso-splash-screen__message"
        >
           
          1
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkbenchPageRender should render WorkbenchPageRender workbench states 3`] = `
<div
  className="vsonline-splash-screen-main "
>
  <div
    className="vsonline-splash-screen-extensions-pane"
  />
  <div
    className="vsonline-splash-screen-tree-pane"
  />
  <div
    className="vsonline-splash-screen-editor"
  >
    <div
      className="vsonline-splash-screen-titlebar"
    >
      <div
        className="vsonline-splash-screen-titlebar-tab"
      />
    </div>
    <div
      className="vsonline-splash-screen-body"
    >
      <div
        className="vso-splash-screen__block"
      >
        <div
          className="vso-splash-screen__message"
        >
          <div
            className="vso-ms-spinner vso-ms-spinner--stopped "
          >
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--1"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--2"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--3"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--4"
            />
          </div>
           
          Getting things ready...
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkbenchPageRender should render WorkbenchPageRender workbench states 4`] = `
<div
  className="vsonline-splash-screen-main "
>
  <div
    className="vsonline-splash-screen-extensions-pane"
  />
  <div
    className="vsonline-splash-screen-tree-pane"
  />
  <div
    className="vsonline-splash-screen-editor"
  >
    <div
      className="vsonline-splash-screen-titlebar"
    >
      <div
        className="vsonline-splash-screen-titlebar-tab"
      />
    </div>
    <div
      className="vsonline-splash-screen-body"
    >
      <div
        className="vso-splash-screen__block"
      >
        <div
          className="vso-splash-screen__message"
        >
          <div
            className="vso-ms-spinner vso-ms-spinner--stopped "
          >
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--1"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--2"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--3"
            />
            <div
              className="vso-ms-spinner__rect vso-ms-spinner__rect--4"
            />
          </div>
           
          Getting things ready...
        </div>
      </div>
    </div>
  </div>
</div>
`;
